libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp

```diff
@@ -155,7 +155,7 @@ uint8_t AP_BattMonitor_SMBus_Generic::read_block(uint8_t reg, uint8_t* data) con
 
     // check PEC
     if (_pec_supported) {
-        uint8_t pec = get_PEC(AP_BATTMONITOR_SMBUS_I2C_ADDR, reg, true, buff, bufflen+1);
+        uint8_t pec = get_PEC(_address, reg, true, buff, bufflen+1);
```

Comment:
```suggestion
        uint8_t pec = get_PEC(_dev->get_bus_address(), reg, true, buff, bufflen+1);
```

In the general case these should be the same, but this will actually keep the battery monitor that was allocated working if you change the address parameter at runtime, because otherwise we would be using the wrong address compared to the device we actually just read data from.

Reply:
Fixed.

Though the driver's address gets allocated on construction so currently there no way I see to change the address at runtime? 

Unless you do _dev->set_address() somewhere

Reply:
>Though the driver's address gets allocated on construction so currently there no way I see to change the address at runtime?
>
> Unless you do _dev->set_address() somewhere

That's actually my point. `_address` is the user facing parameter, which can be changed at runtime, but the actual driver will still use the one it was allocated with. By making the call we stay consistent with what we allocated and are using.

Reply:
durr of course :)